src/stream/src/from_proto/stream_cdc_scan.rs

```diff
@@ -35,13 +35,9 @@ impl ExecutorBuilder for StreamCdcScanExecutorBuilder {
         params: ExecutorParams,
         node: &Self::Node,
         state_store: impl StateStore,
-        stream: &mut LocalStreamManagerCore,
+        _stream: &mut LocalStreamManagerCore,
     ) -> StreamResult<BoxedExecutor> {
         let [upstream]: [_; 1] = params.input.try_into().unwrap();
-        // For reporting the progress.
-        let progress = stream
-            .context
-            .register_create_mview_progress(params.actor_context.id);
```

Comment:
The table stream job created by `CREATE TABLE` should finish immediately (by design), so I think there is no reason to track its progress, right?

Reply:
That make senses.

But from the user's perspective, the percentage shows the progress of catching up to the latest changes, which still sounds to be a very useful metric. Personally I prefer to keep it for monitoring purpose.

BTW this also applies to sink & sink-to-table cc. @kwannoel Any opinions?

Reply:
From user's perspective, I don't really think it matters. Since for non-mviews it will only return once the stream job is created.

For sink and sink to table it indeed still matters and we should still report the progress.
But that is tracked by the `backfill` in `sink` rather than by `table` part. So it's independent of the changes here.

Reply:
For create table with from source, we should also support progress IMO. Since it's long running.

Reply:
I think for create table with source, we can use cdc backfill to track and report the progress.

Reply:
> The table stream job created by CREATE TABLE should finish immediately (by design), so I think there is no reason to track its progress, right?

It finishes "creation" immediately, but in terms of catching up with upstream it may not be yet.
I think what @fuyufjh is pointing out is that we should still track progress, so we know the lag between the table and the upstream source.

Reply:
> the percentage shows the progress of catching up to the latest changes, which still sounds to be a very useful metric

I agree it is a good to have feature to improve user experience. But we need additional efforts to support this progress tracking for cdc backfill, since we don't have statistics for external table on hand. The progress in code is just a placeholder..

Reply:
:thinking: After the new backfill implementation, the table does not backfill the data from the CDC stream which means the table does not backfill by the order of operations anymore... So the data of the table during backfilling could be more meaningless. Should we make the table invisible during the backfilling just like MV's backfilling?

Reply:
> the table does not backfill the data from the CDC stream

What do you mean? 👀

Reply:
@st1page I think we should block the access to the table-on-source table that under backfilling to ensure the consistency in our best. Will support in the foreground mode first. cc @fuyufjh @hzxa21

Reply:
> > the table does not backfill the data from the CDC stream
>
> What do you mean? 👀

I mean that currently the backfill is just like MV on MV's backfill instead of simply chaining the table's data and upstream's changes.

Reply:
> @st1page I think we should block the access to the table-on-source table that under backfilling to ensure the consistency in our best. Will support in the foreground mode first. cc @fuyufjh @hzxa21

From the perspective of consistency, either is okay.

From the perspective of user experience, returning immediately and backfilling in background seems to make users feel at ease.

So I slightly prefer to do the backfilling in background.